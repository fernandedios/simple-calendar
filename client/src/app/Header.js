import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

class Header extends Component {
    renderButtons() {
        let ul = (
            <ul className="navbar-nav ml-auto">
                <li className="nav-item">
                    <Link className="nav-link" to="/">Login</Link>
                </li>
                <li className="nav-item">
                    <Link className="nav-link" to="/signup">Sign Up</Link>
                </li>
            </ul>
        );

        if (this.props.auth) {
            if (this.props.auth.success) {
                ul = (
                    <ul className="navbar-nav ml-auto">
                        <li className="nav-item">
                            <Link className="nav-link" to="/">View Schedule</Link>
                        </li>
                        <li className="nav-item">
                            <Link className="nav-link" to="/add">Add Schedule</Link>
                        </li>
                        <li className="nav-item">
                            <Link className="nav-link" to="/logout">Logout</Link>
                        </li>
                    </ul>
                );
            }
        }
        
        return ul;
    }
    render() {
        return (
            <nav className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top">
                <Link className="navbar-brand" to="/">Simple Calendar</Link>
                <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNav">
                    {this.renderButtons()}
                </div>
            </nav>
        );
    }
}

const mapStateToProps = ({ auth }) => {
    return { auth };
};


export default connect(mapStateToProps)(Header);