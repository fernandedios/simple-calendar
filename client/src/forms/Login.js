import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from './FormActions';

import { TextInput, Button } from './common';

class Login extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email: '',
            password: '',
            error: ''
        };

        this.handleEvent = this.handleEvent.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }

    componentWillMount() {
        if(this.props.auth) {
            if (this.props.auth.success) {
                this.props.history.push('/view');
            }
        }
    }

    componentWillUpdate(nextProps) {
        if(nextProps.auth) {
            if(nextProps.auth.success) {
                this.props.history.push('/view');
            }
        }
    }

    handleEvent(e) {
        e.preventDefault();
        this.setState({ [e.target.name]: e.target.value });
    }

    onSubmit(e) {
        e.preventDefault();
        const { email, password } = this.state;

        if (email.length >= 4 && password.length >= 4 ) {
            this.setState({ error: '' }, () => {
                this.props.doLogin({ email, password });
            });
        }
        else {
            this.setState({ error: 'Invalid email or password' });
        }
    }

    renderError() {
        let error = this.state.error;
        if (this.props.auth) {
            if (!this.props.auth.success) {
                error = this.props.auth.message;
            }
        }
        return <div className="error text-center">{error}</div>;
    }

    render() {
        return (
            <div className="container forms-container">
                <div className="row">
                    <div className="col-md-12">
                        <div className="row">
                            <div className="col-md-6 mx-auto">
                                <div className="card rounded-0">
                                    <div className="card-header">
                                        <h5 className="mb-0">Login</h5>
                                    </div>
                                    <div className="card-body">
                                        {this.renderError()}
                                        <form id="login" onSubmit={this.onSubmit}>
                                            <TextInput name="email" type="text" label="Email Address" value={this.state.email} onChange={this.handleEvent} />
                                            <TextInput name="password" type="password" label="Password" value={this.state.password} onChange={this.handleEvent} />
                                            <Button type="submit" name="btn-login" label="Login" />
                                        </form>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = ({ auth }) => {
    return { auth };
};

export default connect(mapStateToProps, actions)(Login);
