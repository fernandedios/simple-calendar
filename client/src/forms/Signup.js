import React, { Component } from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

import { TextInput, Button } from './common';

class Signup extends Component {
    constructor(props) {
        super(props);

        this.state = {
            name: '',
            email: '',
            password: '',
            confirm: '',
            error: ''
        };

        this.handleEvent = this.handleEvent.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }

    componentWillMount() {
        if(this.props.auth) {
            if (this.props.auth.success) {
                this.props.history.push('/view');
            }
        }
    }

    handleEvent(e) {
        e.preventDefault();
        this.setState({ [e.target.name]: e.target.value });
    }

    onSubmit(e) {
        e.preventDefault();
        const { name, email, confirm, password } = this.state;

        if (name.length > 4 && this.checkEmail(email) && password.length > 4 && confirm === password) {
            axios.post(`${process.env.REACT_APP_SERVER}/api/signup`, { name, email, password })
                .then((response) => {
                    if (response.data.success) {
                        this.setState({ error: 'Signup Successful!' }, () => {
                            setTimeout(() => {
                                this.props.history.push('/');
                            }, 4000);
                        });   
                    }
                    else {
                        this.setState({ error: response.data.message });   
                    }
                })
                .catch((err) => {
                    this.setState({ error: 'Error signing up' });
                })
        }
        else {
            this.setState({ error: 'Check text fields and try again (minimum 5 characters, valid email address)' });
        }
    }

    renderError() {
        return <div className="error text-center">{this.state.error}</div>;
    }

    checkEmail(value) {
        return value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
    }

    render() {
        return (
            <div className="container forms-container">
                <div className="row">
                    <div className="col-md-12">
                        <div className="row">
                            <div className="col-md-6 mx-auto">
                                <div className="card rounded-0">
                                    <div className="card-header">
                                        <h5 className="mb-0">Sign Up</h5>
                                    </div>
                                    <div className="card-body">
                                        {this.renderError()}
                                        <form onSubmit={this.onSubmit}>
                                            <TextInput name="name" type="text" label="Your Name" value={this.state.name} onChange={this.handleEvent} />
                                            <TextInput name="email" type="text" label="Email Address" value={this.state.email} onChange={this.handleEvent} />
                                            <TextInput name="password" type="password" label="Password" value={this.state.password} onChange={this.handleEvent} />
                                            <TextInput name="confirm" type="password" label="Confirm Password" value={this.state.confirm} onChange={this.handleEvent} />
                                            <Button type="submit" name="btn-signup" label="Sign Up" />
                                        </form>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = ({ auth }) => {
    return { auth };
};


export default connect(mapStateToProps)(Signup);
