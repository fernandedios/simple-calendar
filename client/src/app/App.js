import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';

import Header from './Header';
import Login from '../forms/Login';
import Signup from '../forms/Signup';
import AddSchedule from '../forms/AddSchedule';
import ViewSchedule from '../forms/ViewSchedule';
import Logout from '../forms/Logout';

class App extends Component {
  render() {
    return (
      <div>
        <Header />
        <Switch>
          <Route exact path="/" component={Login} />
          <Route path="/signup" component={Signup} />
          <Route path="/add" component={AddSchedule} />
          <Route path="/view" component={ViewSchedule} />
          <Route path="/logout" component={Logout} />
        </Switch>
      </div>
    );
  }
}

export default App;
