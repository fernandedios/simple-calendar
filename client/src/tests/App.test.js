import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import ReactTestRenderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import App from '../app/App';
import Header from '../app/Header';
import Login from '../forms/Login';
import Signup from '../forms/Signup';

const mockStore = configureStore();
const initialState = {};
let store;

beforeEach(() => {
  store = mockStore(initialState);
});

describe('<App />', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });
});

describe('<Header />', () => {
  it('renders the header', () => {
    const output = mount(
      <MemoryRouter>
        <Header store={store} />
      </MemoryRouter>
    );
    expect(output.find('nav.navbar').length).toEqual(1);
  });

  it('should render correctly', () => {
    const output = ReactTestRenderer.create(
    <MemoryRouter>
      <Header store={store} />
    </MemoryRouter>).toJSON();
    expect(output).toMatchSnapshot();
  });
});

describe('<Login />', () => {
  it('renders the login form', () => {
    const output = mount(
        <Login store={store} />
    );
    expect(output.find('form').length).toEqual(1);
  });

  it('should render correctly', () => {
    const output = ReactTestRenderer.create(<Login store={store} />).toJSON();
    expect(output).toMatchSnapshot();
  });
});

describe('<Signup />', () => {
  it('renders the Signup form', () => {
    const output = mount(
        <Signup store={store} />
    );
    expect(output.find('form').length).toEqual(1);
  });

  it('should render correctly', () => {
    const output = ReactTestRenderer.create(<Signup store={store} />).toJSON();
    expect(output).toMatchSnapshot();
  });
});

