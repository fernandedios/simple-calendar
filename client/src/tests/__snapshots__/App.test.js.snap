// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Simple Calendar
  </a>
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarNav"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarNav"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`<Login /> should render correctly 1`] = `
<div
  className="container forms-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 mx-auto"
        >
          <div
            className="card rounded-0"
          >
            <div
              className="card-header"
            >
              <h5
                className="mb-0"
              >
                Login
              </h5>
            </div>
            <div
              className="card-body"
            >
              <div
                className="error text-center"
              >
                
              </div>
              <form
                id="login"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label>
                    Email Address
                  </label>
                  <input
                    className="form-control form-control-lg rounded-0"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control form-control-lg rounded-0"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="btn btn-success btn-lg"
                  name="btn-login"
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Signup /> should render correctly 1`] = `
<div
  className="container forms-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 mx-auto"
        >
          <div
            className="card rounded-0"
          >
            <div
              className="card-header"
            >
              <h5
                className="mb-0"
              >
                Sign Up
              </h5>
            </div>
            <div
              className="card-body"
            >
              <div
                className="error text-center"
              >
                
              </div>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label>
                    Your Name
                  </label>
                  <input
                    className="form-control form-control-lg rounded-0"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Email Address
                  </label>
                  <input
                    className="form-control form-control-lg rounded-0"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control form-control-lg rounded-0"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Confirm Password
                  </label>
                  <input
                    className="form-control form-control-lg rounded-0"
                    id="confirm"
                    name="confirm"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="btn btn-success btn-lg"
                  name="btn-signup"
                  type="submit"
                >
                  Sign Up
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
